src/components/common/calendar/Calendar.js

```diff
@@ -166,6 +187,61 @@ function CalendarApp() {
         <p>
           {event ? `Event Details: ${event.details}` : 'Something went wrong.'}
         </p>
+        <button onClick={handleEdit}>Edit Event</button>
```

Comment:
The only thing I think I would consider changing is the naming of the handleEdit function here to maybe something like toggleShowEditEventForm. At least to me, 'handleEdit' sounds like it would maybe submit the edit or manipulate form values in some way.